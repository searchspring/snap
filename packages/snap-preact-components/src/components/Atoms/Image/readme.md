# Image

Renders an Image with fallback and rollover functionality. 

## Usage

### src
The required `src` prop specifies the URL of the image to render.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' />
```

### alt
The required `alt` prop is the image `alt` attribute.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' />
```

### fallback
The `fallback` prop specifies the URL of the fallback image to render if the primary image fails to load.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} fallback='https://www.example.com/image.jpg' alt='image' />
```

### hoverSrc
The `hoverSrc` prop specifiesthe URL of the alternative image to display on hover.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} hoverSrc={searchResponse.results.mappings.core.hoverImg} alt='image' />
```


### lazy
The `lazy` prop is used to disable the lazy loading feature. Enabled by default. 

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} lazy={false} alt='image' />
```

### Events

#### onMouseOver
The `onMouseOver` prop allows for a custom callback function when the mouse cursor enters the image.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' onMouseOver={(e)=>{console.log(e)}} />
```

#### onMouseOut
The `onMouseOut` prop allows for a custom callback function when the mouse cursor leaves the image.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' onMouseOut={(e)=>{console.log(e)}} />
```

#### onError
The `onError` prop allows for a custom callback function when the image has encountered an error.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' onError={(e)=>{console.log(e)}} />
```

#### onLoad
The `onLoad` prop allows for a custom callback function when the image has finished loading.

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' onLoad={(e)=>{console.log(e)}} />
```

#### onClick
The `onClick` prop allows for a custom callback function when the image is clicked. 

```jsx
<Image src={searchResponse.results.mappings.core.imageUrl} alt='image' onClick={(e)=>{console.log(e)}} />
```
