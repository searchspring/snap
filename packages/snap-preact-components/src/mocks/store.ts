import { UrlManager, QueryStringTranslator, reactLinker } from '@searchspring/snap-url-manager';

export const controller = {
	config: { id: 'search', settings: { redirects: { enabled: false } } },
	client: {},
	store: 'store',
	urlManager: {
		urlState: { id: ['components-pagination--pagination'], viewMode: ['story'] },
		localState: {},
		mergedState: { id: ['components-pagination--pagination'], viewMode: ['story'] },
		omissions: [],
		translator: { config: { queryParameter: 'q' } },
		watcherPool: { callbacks: [null] },
		prevState: {},
		lastUrl: '?id=components-pagination--pagination&viewMode=story',
	},
	eventManager: { events: { beforeSearch: { functions: [null] }, afterSearch: { functions: [null] } } },
	profiler: { namespace: 'search', profiles: [] },
};

const urlManager = new UrlManager(new QueryStringTranslator(), reactLinker);

// export const urlManager = {
// 	globalState:{},
// 	without:{},
// 	getTranslatorUrl:{},
// 	refresh:{},
// 	state:{},
// 	unpackPathAndState:{},
// 	set:{},
// 	merge: {},
// 	remove: {},
// 	reset: {},
// 	withConfig: {},
// 	withGlobals: {},
// 	getTranslatorConfig: {},
// 	href: {},
// 	detach: {},
// 	subscribe: {},
// 	urlState: { id: ['components-pagination--pagination'], viewMode: ['story'] },
// 	localState: {},
// 	mergedState: { id: ['components-pagination--pagination'], viewMode: ['story'] },
// 	omissions: [],
// 	translator: { config: { queryParameter: 'q' } },
// 	watcherPool: { callbacks: [null] },
// 	prevState: {},
// 	lastUrl: '?id=components-pagination--pagination&viewMode=story',
// 	link: {
// 		href: 'nfujeidko',
// 		onClick: {},
// 	},
// 	go: () => {},

// };

export const sorting = {
	controller,
	current: {
		direction: 'desc',
		field: 'sales_rank',
		label: 'Most Popular',
		type: 'field',
		url: urlManager,
	},
	options: [
		{
			direction: 'desc',
			field: 'sales_rank',
			label: 'Most Popular',
			type: 'field',
			url: urlManager,
		},
		{
			direction: 'asc',
			field: 'title',
			label: 'Name (A - Z)',
			type: 'field',
			url: urlManager,
		},
		{
			direction: 'asc',
			field: 'title',
			label: 'Name (Z - A)',
			type: 'field',
			url: urlManager,
		},
		{
			direction: 'desc',
			field: 'price',
			label: 'Price ($$$ - $)',
			type: 'field',
			url: urlManager,
		},
		{
			direction: 'asc',
			field: 'price',
			label: 'Price ($ - $$$)',
			type: 'field',
			url: urlManager,
		},
	],
};

export const filters = [
	{
		facet: {
			field: 'size',
			label: 'Size',
		},
		label: 'Size: 7',
		url: {},
		value: {
			value: '7',
			label: '7',
		},
	},
	{
		facet: {
			field: 'size',
			label: 'Size',
		},
		label: 'Size: 7.5',
		url: {},
		value: {
			value: '7.5',
			label: '7.5',
		},
	},
	{
		facet: {
			field: 'color',
			label: 'Color',
		},
		label: 'Color: Blue',
		url: {},
		value: {
			value: 'Blue',
			label: 'Blue',
		},
	},
];

// export const paginationFirstPageMock: PaginationStore = {
// 	controller,
// 	page: 1,
// 	pageSize: 30,
// 	totalResults: 1305,
// 	pageSizeOptions: [
// 		{ label: 'Show 24', value: 24 },
// 		{ label: 'Show 48', value: 48 },
// 		{ label: 'Show 72', value: 72 },
// 	],
// 	begin: 1,
// 	current: {
// 		url: urlManager,
// 		active: true,
// 		number: 1,
// 		key: '1',
// 		services: {
// 			urlManager: urlManager
// 		},
// 	},
// 	first: {
// 		url: urlManager,
// 		active: true,
// 		number: 1,
// 		key: '1',
// 		services: {
// 			urlManager: urlManager
// 		},
// 	},
// 	last: {
// 		url: urlManager,
// 		active: false,
// 		number: 44,
// 		key: '44',
// 		services: {
// 			urlManager: urlManager
// 		},
// 	},
// 	next: {
// 		url: urlManager,
// 		active: false,
// 		number: 2,
// 		key: '2',
// 		services: {
// 			urlManager: urlManager
// 		},
// 	},
// 	end: 30,
// 	totalPages: 44,
// 	multiplePages: true,
// 	setPageSize: (num: number) => {
// 		if (num) {
// 			return urlManager.remove('page').set('pageSize', num).go();
// 		}
// 	},

// 	getPages: (min:number | undefined, max: number | undefined) => {
// 		return [
// 			{
// 				active: true,
// 				number: 1,
// 				url: urlManager,
// 				key: '1',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 2,
// 				url: urlManager,
// 				key: '2',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 3,
// 				url: urlManager,
// 				key: '3',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 4,
// 				url: urlManager,
// 				key: '4',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 5,
// 				url: urlManager,
// 				key: '5',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 		];
// 	},
// };

// export const paginationMock: PaginationStore = {
// 	services: {
// 		urlManager: urlManager
// 	},
// 	defaultPageSize: 30,
// 	infinite: false,
// 	controllerConfig: controller.config,

// 	page: 5,
// 	pageSize: 30,
// 	totalResults: 1305,
// 	pageSizeOptions: [
// 		{ label: 'Show 24', value: 24 },
// 		{ label: 'Show 48', value: 48 },
// 		{ label: 'Show 72', value: 72 },
// 	],
// 	begin: 1,
// 	current: {
// 		url: urlManager,
// 		active: true,
// 		number: 5,
// 		key: '5',
// 		services: {
// 			urlManager: urlManager
// 		}
// 	},
// 	first: {
// 		url: urlManager,
// 		active: false,
// 		number: 1,
// 		key: '1',
// 		services: {
// 			urlManager: urlManager
// 		}
// 	},
// 	last: {
// 		url: urlManager,
// 		active: false,
// 		number: 44,
// 		key: '44',
// 		services: {
// 			urlManager: urlManager
// 		}
// 	},
// 	previous: {
// 		url: urlManager,
// 		active: false,
// 		number: 4,
// 		key: '4',
// 		services: {
// 			urlManager: urlManager
// 		}
// 	},
// 	next: {
// 		url: urlManager,
// 		active: false,
// 		number: 6,
// 		key: '6',
// 		services: {
// 			urlManager: urlManager
// 		}
// 	},
// 	end: 30,
// 	totalPages: 44,
// 	multiplePages: true,
// 	setPageSize: (num: number) => {
// 		if (num) {
// 			return urlManager.remove('page').set('pageSize', num).go();
// 		}
// 	},
// 	getPages: (min:number | undefined, max: number | undefined) => {
// 		return [
// 			{
// 				active: false,
// 				number: 3,
// 				url: urlManager,
// 				key: '3',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 4,
// 				url: urlManager,
// 				key: '4',
// 				services: {
// 						urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: true,
// 				number: 5,
// 				url: urlManager,
// 				key: '5',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 6,
// 				url: urlManager,
// 				key: '6',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 			{
// 				active: false,
// 				number: 7,
// 				url: urlManager,
// 				key: '7',
// 				services: {
// 					urlManager: urlManager
// 				}
// 			},
// 		];
// 	},
// };
