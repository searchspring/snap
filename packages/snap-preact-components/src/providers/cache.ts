export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
