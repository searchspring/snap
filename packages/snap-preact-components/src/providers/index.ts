export * from './store';
export * from './theme';
