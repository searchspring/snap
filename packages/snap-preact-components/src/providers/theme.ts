import { ThemeProvider as EmotionTheme } from '@emotion/react';
export { css, useTheme, withTheme } from '@emotion/react';

interface ThemeProviderProps {
	theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
	children?: any;
}
interface ThemeProvider {
	(props: ThemeProviderProps): React.ReactElement;
}
export const ThemeProvider: ThemeProvider = EmotionTheme;

export const defaultTheme: Theme = {
	colors: {
		primary: '#3A23AD',
		secondary: '#00cee1',
		hover: '#f8f6fd',
		text: {
			secondary: '#ffffff',
		},
		message: {
			error: '#cc1212',
			warning: '#ecaa15',
			info: '#4c3ce2',
		},
	},
	components: {},
};
export interface Theme {
	variables?: {
		[key: string]: unknown;
	};
	colors?: {
		primary: string;
		secondary: string;
		hover?: string;
		text?: {
			secondary: string;
		};
		message?: {
			error: string;
			warning: string;
			info: string;
		};
	};
	components?: any;
	namedComponents?: any;
}
