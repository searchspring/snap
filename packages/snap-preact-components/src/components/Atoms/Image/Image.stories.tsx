import { h } from 'preact';

import { ArgsTable, PRIMARY_STORY } from '@storybook/addon-docs/blocks';

import { Image, FALLBACK_IMAGE_URL } from './Image';
import { componentArgs } from '../../../utilities';
import { searchResponse } from '../../../mocks/searchResponse';
import Readme from '../Image/readme.md';

export default {
	title: `Atoms/Image`,
	component: Image,
	parameters: {
		docs: {
			page: () => (
				<div>
					<Readme />
					<ArgsTable story={PRIMARY_STORY} />
				</div>
			),
		},
	},
	decorators: [
		(Story) => (
			<div
				style={{
					maxWidth: '300px',
				}}
			>
				<Story />
			</div>
		),
	],
	argTypes: {
		src: {
			description: 'Image url',
			type: { required: true },
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'text' },
		},
		alt: {
			description: 'Image alt text',
			type: { required: true },
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'text' },
		},
		fallback: {
			description: 'Fallback image url',
			defaultValue: FALLBACK_IMAGE_URL,
			table: {
				type: {
					summary: 'string',
				},
				defaultValue: { summary: 'string' },
			},
			control: { type: 'text' },
		},
		hoverSrc: {
			description: 'Image onHover url',
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'text' },
		},
		maxHeight: {
			description: 'Image max height',
			defaultValue: '320px',
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'text' },
		},
		onLoad: {
			description: 'Image loaded event handler',
			table: {
				type: {
					summary: 'function',
				},
			},
			action: 'onLoad',
		},
		onClick: {
			description: 'Image click event handler',
			table: {
				type: {
					summary: 'function',
				},
			},
			action: 'onClick',
		},
		onMouseOver: {
			description: 'Image mouse enter event handler',
			table: {
				type: {
					summary: 'function',
				},
			},
			action: 'onMouseOver',
		},
		onMouseOut: {
			description: 'Image mouse exit event handler',
			table: {
				type: {
					summary: 'function',
				},
			},
			action: 'onMouseOut',
		},
		...componentArgs,
	},
};

const Template = (args) => <Image {...args} style={{ width: '100%' }} />;

export const Default = Template.bind({});
Default.args = {
	src: searchResponse.results[6].mappings.core.imageUrl,
	alt: searchResponse.results[6].mappings.core.name,
};

export const BrokenImg = Template.bind({});
BrokenImg.args = {
	src: 'intentionally_broken_image.jpg',
	alt: searchResponse.results[6].mappings.core.name,
};

export const ManualFallBack = Template.bind({});
ManualFallBack.args = {
	src: 'intentionally_broken_image.jpg',
	alt: searchResponse.results[5].mappings.core.name,
	fallback: searchResponse.results[5].mappings.core.imageUrl,
};

export const onhover = Template.bind({});
onhover.args = {
	src: searchResponse.results[6].mappings.core.imageUrl,
	alt: searchResponse.results[6].mappings.core.name,
	hoverSrc: searchResponse.results[7].mappings.core.imageUrl,
};
