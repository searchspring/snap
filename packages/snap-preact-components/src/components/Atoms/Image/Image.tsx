/** @jsx jsx */
import { h } from 'preact';

import { useState } from 'preact/hooks';
import { jsx, css } from '@emotion/react';
import classnames from 'classnames';

import { Theme, useTheme } from '../../../providers';
import { ComponentProps } from '../../../types';

export const FALLBACK_IMAGE_URL = '//cdn.searchspring.net/ajax_search/img/default_image.png';

const CSS = {
	image: ({ visibility, style }) =>
		css({
			visibility,
			...style,
		}),
};

export function Image(properties: ImageProps): JSX.Element {
	const globalTheme: Theme = useTheme();

	const props: ImageProps = {
		// default props
		fallback: FALLBACK_IMAGE_URL,
		// global theme
		...globalTheme?.components?.image,
		// props
		...properties,
		...properties.theme?.components?.image,
	};

	const { alt, src, fallback, hoverSrc, onMouseOver, onMouseOut, onLoad, onClick, disableStyles, className, style } = props;

	const [visibility, setVisibility] = useState('hidden');
	const [isHovering, setHover] = useState(false);

	return (
		<img
			css={!disableStyles && CSS.image({ visibility, style })}
			className={classnames('ss__image', className)}
			src={(isHovering ? hoverSrc : src) || fallback}
			alt={alt}
			title={alt}
			loading="lazy"
			onLoad={() => {
				setVisibility('visible');
				onLoad && onLoad();
			}}
			onClick={(e) => onClick && onClick(e as any)}
			onError={(e) => ((e.target as HTMLImageElement).src = fallback)}
			onMouseOver={(e) => {
				hoverSrc && setHover(true);
				onMouseOver && onMouseOver(e as any);
			}}
			onMouseOut={(e) => {
				hoverSrc && setHover(false);
				onMouseOut && onMouseOut(e as any);
			}}
		/>
	);
}

export interface ImageProps extends ComponentProps {
	alt: string;
	src: string;
	fallback?: string;
	hoverSrc?: string;
	onMouseOver?: (e: MouseEvent) => void;
	onMouseOut?: (e: MouseEvent) => void;
	onLoad?: () => void;
	onClick?: (e: MouseEvent) => void;
}
