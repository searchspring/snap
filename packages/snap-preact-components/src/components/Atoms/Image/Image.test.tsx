import { h } from 'preact';

import { ThemeProvider } from '../../../providers/theme';
import { render } from '@testing-library/preact';

import { Image, FALLBACK_IMAGE_URL } from './Image';
import { searchResponse, badSearchResponse } from '../../../mocks/searchResponse';

describe('image Component', () => {
	const result = searchResponse.results[0].mappings.core;
	const badResult = badSearchResponse.results[0].mappings.core;

	it('renders', () => {
		const rendered = render(<Image alt={result.name} src={result.thumbnailImageUrl} />);
		const imageElement = rendered.container.querySelector('.ss-image');
		expect(imageElement).toBeInTheDocument();
		expect(imageElement).toHaveAttribute('src', result.thumbnailImageUrl);
		expect(imageElement).toHaveAttribute('alt', result.name);
		expect(imageElement).toHaveAttribute('title', result.name);

		expect(imageElement.classList).toHaveLength(1);
	});

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<Image className={className} alt={result.name} src={result.thumbnailImageUrl} />);

		const imageElement = rendered.container.querySelector('.ss-image');

		expect(imageElement).toHaveClass(className);
	});

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<Image className={className} alt={result.name} src={result.thumbnailImageUrl} />);
		const imageElement = rendered.container.querySelector('.ss-image');

		expect(imageElement).toHaveClass(className);
	});

	it('disables styles', () => {
		const rendered = render(<Image disableStyles alt={result.name} src={result.thumbnailImageUrl} />);
		const imageElement = rendered.container.querySelector('.ss-image');

		expect(imageElement.classList).toHaveLength(1);
	});

	describe('Working Image', () => {
		it('renders image', () => {
			const rendered = render(<Image alt={result.name} src={result.thumbnailImageUrl} />);
			const imageElement = rendered.container.querySelector('.ss-image');
			expect(imageElement).toBeInTheDocument();
			expect(imageElement).toHaveAttribute('src', result.thumbnailImageUrl);
		});
	});

	describe('Broken Image', () => {
		it('should display default fallback image', () => {
			const rendered = render(<Image alt={badResult.name} src={badResult.thumbnailImageUrl} />);
			const imageElement = rendered.container.querySelector('.ss-image');
			expect(imageElement).toHaveAttribute('src', FALLBACK_IMAGE_URL);
		});

		it('should display custom fallback image', () => {
			const fallbackImage = 'https://www.telegraph.co.uk/content/dam/Pets/spark/royal-canin/happy-puppy-xlarge.jpg?imwidth=1200';
			const rendered = render(<Image alt={badResult.name} src={badResult.thumbnailImageUrl} fallback={fallbackImage} />);
			const imageElement = rendered.container.querySelector('.ss-image');
			expect(imageElement).toHaveAttribute('src', fallbackImage);
		});
	});
});

describe('Image theming works', () => {
	const result = searchResponse.results[0].mappings.core;

	it('is themeable with ThemeProvider', () => {
		const globalTheme = {
			components: {
				image: {
					disableStyles: true,
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={globalTheme}>
				<Image alt={result.name} src={result.thumbnailImageUrl} />
			</ThemeProvider>
		);
		const pagination = rendered.container.querySelector('.ss-image');
		expect(pagination).toBeInTheDocument();
		expect(pagination.classList.length).toBe(1);
	});

	it('is themeable with theme prop', () => {
		const propTheme = {
			components: {
				image: {
					disableStyles: true,
				},
			},
		};
		const rendered = render(<Image alt={result.name} src={result.thumbnailImageUrl} theme={propTheme} />);
		const pagination = rendered.container.querySelector('.ss-image');
		expect(pagination).toBeInTheDocument();
		expect(pagination.classList.length).toBe(1);
	});

	it('is theme prop overrides ThemeProvider', () => {
		const globalTheme = {
			components: {
				image: {
					disableStyles: false,
				},
			},
		};
		const propTheme = {
			components: {
				image: {
					disableStyles: true,
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={globalTheme}>
				<Image alt={result.name} src={result.thumbnailImageUrl} theme={propTheme} />
			</ThemeProvider>
		);

		const pagination = rendered.container.querySelector('.ss-image');
		expect(pagination).toBeInTheDocument();
		expect(pagination.classList.length).toBe(1);
	});
});
