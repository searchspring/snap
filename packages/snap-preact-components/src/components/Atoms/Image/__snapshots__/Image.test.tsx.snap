// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image theming works is theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__image"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="//cdn.searchspring.net/ajax_search/img/default_image.png"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`Image theming works is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__image"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="//cdn.searchspring.net/ajax_search/img/default_image.png"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`Image theming works is themeable with theme prop 1`] = `
<DocumentFragment>
  <div
    class="ss__image"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="//cdn.searchspring.net/ajax_search/img/default_image.png"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component Broken Image should display custom fallback image 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="https://www.telegraph.co.uk/content/dam/Pets/spark/royal-canin/happy-puppy-xlarge.jpg?imwidth=1200"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component Broken Image should display default fallback image 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="//cdn.searchspring.net/ajax_search/img/default_image.png"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component Working Image renders image 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Freezing Point Cream Turtleneck Sweater"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Freezing Point Cream Turtleneck Sweater"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component click func custom onclick src on hover 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component click func custom onclick src on hover 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component disables styles 1`] = `
<DocumentFragment>
  <div
    class="ss__image"
  >
    <img
      alt="Freezing Point Cream Turtleneck Sweater"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Freezing Point Cream Turtleneck Sweater"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component hover src should change src on hover & run a custom onhoverfunc prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component hover src should change src on hover & run a custom onhoverfunc prop 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Play For Keeps Burgundy Velvet Bralette"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/478a3650_thumb_med.jpg"
      title="Play For Keeps Burgundy Velvet Bralette"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Freezing Point Cream Turtleneck Sweater"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Freezing Point Cream Turtleneck Sweater"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component renders with classname 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image classy emotion-0"
  >
    <img
      alt="Freezing Point Cream Turtleneck Sweater"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Freezing Point Cream Turtleneck Sweater"
    />
  </div>
</DocumentFragment>
`;

exports[`image Component uses bocachica theme 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: auto;
}

.emotion-0 img {
  visibility: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
}

<div
    class="ss__image emotion-0"
  >
    <img
      alt="Freezing Point Cream Turtleneck Sweater"
      loading="lazy"
      src="https://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_thumb_med/110_lcp_9262_copyright_loganpotterf_2016_thumb_med.jpg"
      title="Freezing Point Cream Turtleneck Sweater"
    />
  </div>
</DocumentFragment>
`;
